core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -80,13 +80,13 @@ object Defaults {
   val LogCleanupPolicy = Delete
   val LogCleanerThreads = 1
   val LogCleanerIoMaxBytesPerSecond = Double.MaxValue
-  val LogCleanerDedupeBufferSize = 500 * 1024 * 1024L
+  val LogCleanerDedupeBufferSize = 128 * 1024 * 1024L
   val LogCleanerIoBufferSize = 512 * 1024
   val LogCleanerDedupeBufferLoadFactor = 0.9d
   val LogCleanerBackoffMs = 15 * 1000
   val LogCleanerMinCleanRatio = 0.5d
-  val LogCleanerEnable = false
-  val LogCleanerDeleteRetentionMs = 24 * 60 * 60 * 1000L
+  val LogCleanerEnable = true
+  val LogCleanerDeleteRetentionMs = 7 * 24 * 60 * 60 * 1000L
```

Comment:
Is there a particular reason to change this? This is really to just control how long we keep the delete tombstone. Basically, if a consumer has seen a deleted key, it's expected to consume the delete marker within this time. 1 day seems enough.